test.md

```diff
@@ -258,6 +295,35 @@ We allow to `invoke` a function by its exported name in the test code.
          <exports> ... ENAME |-> TFIDX ... </exports>
 ```
 
+### Registering Modules
+
+We first define how modules get stored away.
+
+```k
+    rule <k> #storeInstance => . ... </k>
+         <moduleInst> INST </moduleInst>
+         <nextModuleIdx> NEXT => NEXT +Int 1 </nextModuleIdx>
+         <moduleInstances> ... .Map => NEXT |-> INST ... </moduleInstances>
+```
+
+We will reference modules by name in imports.
+`register` is the instruction that allows us to associate a name with a module.
+
+```k
+    syntax Auxil ::= "(" "register" String            ")"
+                   | "(" "register" String Identifier ")"
+                   |     "register" String Int
```

Comment:
Can we just use `"(" "register" String TextFormatIdx ")"` for this rule rather than separately declaring it for `Identifier` and `Int`?

Reply:
Sure, that makes sense.